src/utils/convertCurrency.mjs

```diff
@@ -1,18 +1,21 @@
-function replaceCommasWithDots(stringAmount) {
-  return stringAmount.replace(',', '.');
-}
-
-export function calculateTargetCurrencyAmount(sourceCurrencyNominal, exchangeRate, sourceCurrencyAmount) {
-  if (!sourceCurrencyNominal && !exchangeRate) return null;
-  if (!sourceCurrencyNominal && !sourceCurrencyAmount) return sourceCurrencyNominal * exchangeRate;
+import { replaceCommasWithDots } from "./currency.mjs";
+import { DEFAULT_CURRENCY } from "../constants/index.mjs";
 
+function calculateTargetCurrencyToDefaultCurrency(exchangeRate, sourceCurrencyAmount) {
   return exchangeRate * sourceCurrencyAmount;
 }
 
-export function convertAmountToTargetCurrency({ sourceCurrencyNominal, exchangeRate, sourceCurrencyAmount }) {
-  return calculateTargetCurrencyAmount(
-    sourceCurrencyNominal,
+function calculateDefaultCurrencyToTargetCurrency(exchangeRate, sourceCurrencyAmount) {
+  return parseFloat((sourceCurrencyAmount / exchangeRate).toFixed(2))
+}
+
+export function convertTargetCurrency({ exchangeRate, sourceCurrencyAmount, currencyCode }) {
+  return currencyCode == DEFAULT_CURRENCY ? calculateTargetCurrencyToDefaultCurrency(
     parseFloat(replaceCommasWithDots(exchangeRate)),
     parseFloat(replaceCommasWithDots(sourceCurrencyAmount))
-  );
+  ): calculateDefaultCurrencyToTargetCurrency(
+    parseFloat(replaceCommasWithDots(exchangeRate)),
+    parseFloat(replaceCommasWithDots(sourceCurrencyAmount))
+  )
 }
```

Comment:
This can look like:
```typescript
export function prepareNumber = (stringifiedNumber: string) => {
  parseFloat(replaceCommasWithDots(stringifiedNumber);
};

export function convertTargetCurrency({ exchangeRate, sourceCurrencyAmount, currencyCode }) {|
  const preparedExchangeRate = prepareNumber(exchangeRate);
  const preparedSourceCurrencyAmount = prepareNumber(sourceCurrencyAmount);
  
  return currencyCode == DEFAULT_CURRENCY ? calculateTargetCurrencyToDefaultCurrency(
      preparedExchangeRate,
      preparedSourceCurrencyAmount,
    ) : calculateDefaultCurrencyToTargetCurrency(
      preparedExchangeRate,
      preparedSourceCurrencyAmount,
    )
}
```